meveo-admin/ejbs/src/main/java/org/meveo/service/base/BusinessService.java

```diff
@@ -169,6 +179,87 @@ public BusinessEntity findByEntityClassAndCode(Class<?> clazz, String code) {
         }
     }
 
+    /** 
+     * Find which the module the entity belongs to 
+
+     * @param entity
+     * @return MeveoModule, or null otherwise
+     */
+    @SuppressWarnings({ "rawtypes", "deprecation" })
+	public MeveoModule findModuleOf(P entity){
+    	MeveoModule module = null;
+    	if (entity != null) {
+    		Session session = this.getEntityManager().unwrap(Session.class);
+    		Query q = session.createQuery("SELECT m.module FROM ModuleItem m WHERE m.code = :code AND m.ItemType = :itemType");
+    		q.setString("code", entity.getCode());
+    		q.setString("itemType", entity.getClass().getName());
+    		module = (MeveoModule) q.getResultList().get(0);
+    	}
+    	return module;
+    }
+    
+    /**
+     * Remove the entity that belongs to the module
+     * 
+     * @param entity
+     * @param module
+     */
+    public void removeFilesFromModule(P entity, MeveoModule module) {
+    	MeveoModuleDto meveoModuleDto = new MeveoModuleDto(module);
+    	BusinessEntityDto businessEntityDto = new BusinessEntityDto(entity);
+    	File fileToDelete = new File("..\\git\\"+meveoModuleDto.getCode()+"\\"+businessEntityDto.getClass().getAnnotation(ModuleItem.class).path()+"\\"+entity.getCode()+".json");
+
+    	fileToDelete.delete();
+    }
+    
+    /**
+     * Create the entity in the dedicated module
+     * 
+     * @param entity
+     * @param module
+     */
+    public void addFilesToModule(P entity, MeveoModule module) {
+    	MeveoModuleDto meveoModuleDto = new MeveoModuleDto(module);
+    	BusinessEntityDto businessEntityDto = new BusinessEntityDto(entity);
+    	String businessEntityDtoSerialize = JacksonUtil.toString(businessEntityDto);
```

Comment:
This part will need to be reviewed as it won't work, but I will take care of that part as it's a little tricky. I'll explain you what I did